webui/react/src/hooks/useCheckpointFlow.tsx

```diff
@@ -83,18 +85,25 @@ export const useCheckpointFlow = ({
   }, [canceler.signal]);
 
   useEffect(() => {
-    fetchModels();
-  }, [fetchModels]);
+    if (models.isNotLoaded) fetchModels();
+  }, [fetchModels, models.isNotLoaded]);
```

Comment:
this is bordering on refactoring but i think it'd be better if the `models` prop were reactive, given that we're passing a loadable.

we should be able to do this by switching the `models` const from independent state to just using the `useAsync` hook:
```tsx
  const models = useAsync(async (canceler) => {
    if (modelsIn) return modelsIn;
    try {
      const response = await getModels(
        {
          archived: false,
          orderBy: 'ORDER_BY_DESC',
          sortBy: validateDetApiEnum(
            V1GetModelsRequestSortBy,
            V1GetModelsRequestSortBy.LASTUPDATEDTIME,
          ),
        },
        { signal: canceler.signal },
      );
      return response.models;
    } catch (e) {
      handleError(e, {
        publicSubject: 'Unable to fetch models.',
        silent: true,
        type: ErrorType.Api,
      });
      return NotLoaded;
    }
  }, [modelsIn]);
```

note that this assumes that we're removing the default value for the prop. while we're in here, we can also remove the abort controller state in the hook, because it's never actually cancelled anywhere.

Reply:
Combined this with existing `useFetchModels` hook.

Reply:
nice!